### PR TITLE
Helmet router material

### PR DESCRIPTION
**Pull request by rymaju**
Added Helmet, and some example usage of React Router DOM, Helmet, and Material UI.

A lot of stuff was left out to keep things vague enough for general use, but this should be a (decent???) starting point.

image
image

### CODE REVIEW

**Comment by rymaju** [comment]
Decided not to use react router sitemap, it would require moving the Router logic intro a separate jsx file. I think that thats a bit too messy from an organizational standpoint. We could just write our own sitemap.xml if we wanted to.

File: src/App.tsx [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
```
Comment (jackblanc) [inline src/App.tsx]: you could use a fragment here `<> {content} </>`

File: src/App.tsx [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+      <Helmet>
+        <meta
+          name='keywords'
+          content='C4C,code,for,community,code4community,codeforcommunity,northeastern,boston'
```
Comment (jackblanc) [inline src/App.tsx:17]: how do spaces affect this?

File: src/components/ExamplePage.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  Container,
+  Box,
+  Typography,
+  Paper,
+  Card,
+  CardContent,
+  CardActions,
+  Button
+} from '@material-ui/core'
+function ExamplePage () {
```
Comment (jackblanc) [inline src/components/ExamplePage.tsx]: I generally prefer `const ExamplePage = () => {` as opposed to this syntax, though I'm not sure that there's any technical difference

File: src/components/ExamplePage.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import {
+  Container,
+  Box,
+  Typography,
+  Paper,
+  Card,
+  CardContent,
+  CardActions,
+  Button
+} from '@material-ui/core'
```
Comment (jackblanc) [inline src/components/ExamplePage.tsx:12]: There should be a space here, which would be something caught by tslint, is that set up in this repo? I didn't do anything to do so, curious if it just works out of box

File: src/components/ExamplePage.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        <Typography variant='h5'>Hey, this is example text size h5.</Typography>
+        <Typography variant='h6'>Hey, this is example text size h6.</Typography>
+        <Typography variant='body1'>
+          Hey, this is example text size body1.
+        </Typography>
+        <Typography variant='body2'>
+          Hey, this is example text size body2.
+        </Typography>
+
+        <br />
```
Comment (jackblanc) [inline src/components/ExamplePage.tsx]: Can we use something other than `<br />`? I feel like that's hard to work with long term

File: src/components/Home.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      </Helmet>
+      <Box my={4}>
+        <Box mb={5}>
+          <Typography variant='h2'>Frontend Scaffold</Typography>
+          <Typography variant='h4'>
+            The scaffold for C4C frontend projects.
+          </Typography>
+
+          <Typography variant='subtitle2' gutterBottom>
+            Using React+Typescript and Material UI. Its probably better than
```
Comment (jackblanc) [inline src/components/Home.tsx]: `It's`

also, remove that sentence 😂

File: src/components/Home.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+          container
+          direction='row'
+          justify='center'
+          alignItems='center'
+          spacing={2}
+        >
+          <Grid item sm>
+            <Typography variant='h6'>Subheading1</Typography>
+            <Typography variant='body1'>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
```
Comment (jackblanc) [inline src/components/Home.tsx:34]: ⛏️  (this means nitpick, not a huge deal)
If you could have different lorem ipsum for each that might look better :)

File: src/components/NotFound.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import { Helmet } from 'react-helmet'
+
+import {
+  Container,
+  Box,
+  Typography,
+
```
Comment (jackblanc) [inline src/components/NotFound.tsx:5]: def need some linting setup for stuff like this extra line

**Review by jackblanc — COMMENTED** [review]
Hey sorry I was doing coop all day. This is a great start! If you could configure linting as a part of this that would be great but not absolutely necessary

File: src/App.tsx (hunk 2 above)
Comment (rymaju) [reply src/App.tsx:17]: The delimiter is just the commas, so adding spaces wouldn't make a difference. I can add spaces to make it prettier if needed.

File: src/components/ExamplePage.tsx (hunk 4 above)
Comment (rymaju) [reply src/components/ExamplePage.tsx:12]: I didn't have the tslint extension when I was coding this so I didn't have any linting, I'll check everything over again!

File: src/components/ExamplePage.tsx (hunk 3 above)
Comment (rymaju) [reply src/components/ExamplePage.tsx]: Thats probably better for "this" scoping, I'll make those changes.

File: src/components/ExamplePage.tsx (hunk 5 above)
Comment (rymaju) [reply src/components/ExamplePage.tsx]: The spacing was just so you can properly see the shadow of the paper/cards which are usually used in a grid with automatic spacing. I can wrap it in a Box with some extra y margin instead.

File: src/App.tsx (hunk 1 above)
Comment (rymaju) [reply src/App.tsx]: Not super sure what you mean here, probably some lack of React knowledge on my part. I'll look it up and try to fix it. Let me know if I get it wrong

**Comment by rymaju** [comment]
- Made components in the form: `const Component = () => { ... }`
- Changed up the lorem ipsum
- Removed snarky comment
- Added spaces properly (still not sure if tslint is set up correctly but supposedly im linting on save)
- removed `<br/>`tags and replaced them with a <Box> wrapper with margin/padding

File: src/App.tsx [hunk 9]
```diff
@@ -8,9 +8,9 @@ import NotFound from './components/NotFound'
 
 import { AppBar, Tabs, Tab } from '@material-ui/core'
 
-function App () {
+const App = () => {
```
Comment (jackblanc) [inline src/App.tsx]: Should these components be typed? 

`const App: React.FC<> = () => {}`

**Review by jackblanc — APPROVED** [review]
Great work! One comment on typing components

**Pull request merged**